modules/core/src/core/attribute.js

```diff
@@ -1,39 +1,52 @@
 /* eslint-disable complexity */
 import GL from '@luma.gl/constants';
-import {Buffer} from '../webgl';
+import {Accessor, Buffer} from '../webgl';
 import {log, uid} from '../utils';
 import {hasFeature, FEATURES} from '../webgl-context/context-features';
 
+const ACCESSOR_DEFAULTS = {
+  offset: 0,
+  stride: 0,
+  normalized: false,
+  integer: false,
+  divisor: 0
+};
+
 export default class Attribute {
-  constructor(gl, opts = {}) {
-    const {id = uid('attribute'), type, isIndexed = false} = opts;
+  constructor(gl, props = {}) {
+    const {id = uid('attribute'), type, isIndexed = false} = props;
 
     // Options that cannot be changed later
     this.gl = gl;
     this.id = id;
+
     this.isIndexed = isIndexed;
     this.target = isIndexed ? GL.ELEMENT_ARRAY_BUFFER : GL.ARRAY_BUFFER;
-    this.type = type;
 
-    if (isIndexed && !type) {
-      // If the attribute is indices, auto infer the correct type
-      // WebGL2 and WebGL1 w/ uint32 index extension support accepts Uint32Array, otherwise Uint16Array
-      this.type =
-        gl && hasFeature(gl, FEATURES.ELEMENT_INDEX_UINT32) ? GL.UNSIGNED_INT : GL.UNSIGNED_SHORT;
-    }
-
-    // Initialize the attribute descriptor, with WebGL and metadata fields
+    this.constant = false;
     this.value = null;
     this.externalBuffer = null;
     this.buffer = null;
     this.userData = {}; // Reserved for application
-    this.update(opts);
+    this.accessor = new Accessor(ACCESSOR_DEFAULTS);
 
-    // Sanity - no app fields on our attributes. Use userData instead.
-    Object.seal(this);
+    this.update(props);
+
+    // Update the attribute accessor
+    const accessorFields = {type};
+    // If the attribute contains indices, auto infer the correct type
+    // WebGL2 and WebGL1 w/ uint32 index extension support accepts Uint32Array, otherwise Uint16Array
+    if (isIndexed && !type) {
+      accessorFields.type =
+        accessorFields.type || (gl && hasFeature(gl, FEATURES.ELEMENT_INDEX_UINT32))
+          ? GL.UNSIGNED_INT
+          : GL.UNSIGNED_SHORT;
+    }
 
-    // Check all fields and generate helpful error messages
-    this._validateAttributeDefinition();
+    this.accessor = new Accessor(this.accessor, accessorFields);
```

Comment:
We have sprawling code in deck.gl that uses the legacy API and this will be hard to catch.

- Check at least one of the deprecated props?
- Check and warn if `accessor` does not exist?

Reply:
Yes, the hard coupling with deck.gl (I believe this class is not just used, it is subclassed) is what has kept me from tackling this in the past, so your input is valued. And I'll need to do a test inside deck.gl before merging this. 

> * Check at least one of the deprecated props?
> * Check and warn if accessor does not exist?

Not quite sure what you are suggesting? Are you suggesting that we make sure that the application overrides at least some accessor prop whether in the main props or props.accessor? 

With the auto-deduction of `size` and `type` from GLSL it is often no longer necessary to specify any accessor props.

My thinking was
* I copy any deprecated props to `this.accessor` on line 33. 
* `props.accessor` does not need to be provided.